sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  Body,
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+} from "@nestjs/common";
+import { ApplicationService } from "../../services";
+import { UserToken } from "../../auth/decorators/userToken.decorator";
+import { IUserToken } from "../../auth/userToken.interface";
+import BaseController from "../BaseController";
+import { CreateApplicationDto } from "./models/application.model";
+
+@Controller("application")
+export class ApplicationController extends BaseController {
+  constructor(private readonly applicationService: ApplicationService) {
+    super();
+  }
+
+  @Get(":id/data")
+  async getByApplicationId(
+    @Param("id") applicationId: string,
+    @UserToken() userToken: IUserToken,
+  ): Promise<any> {
+    const application = await this.applicationService.getApplicationById(
+      applicationId,
+      userToken.userName,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        `Application id ${applicationId} was not found.`,
```

Comment:
Consider rewording the error message to "No Application Found". Dont want to log /give out application id"

Reply:
Please note that the aplication Id is the one received from the user, not an internal one.
We are not disclosing any information here.